src/matrix_multiply_omp_compare/CMakeLists.txt

```diff
@@ -1,44 +1,23 @@
-############################################################################
-#
-#  Copyright (C) Codeplay Software Limited
-#  Licensed under the Apache License, Version 2.0 (the "License");
-#  you may not use this file except in compliance with the License.
-#  You may obtain a copy of the License at
-#
-#      http://www.apache.org/licenses/LICENSE-2.0
-#
-#  Unless required by applicable law or agreed to in writing, software
-#  distributed under the License is distributed on an "AS IS" BASIS,
-#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-#  See the License for the specific language governing permissions and
-#  limitations under the License.
-#
-############################################################################
+add_executable(matrix_multiply_omp_compare matrix-multiply.cpp)
+set(MATRIX_MULTIPLY_FLAGS ${SYCL_FLAGS})
 
-# Set the project name
-project(SYCL-samples_matrix_multiply_omp_compare
-        VERSION 1.0
-        DESCRIPTION "Compares an OpenMP blocked matrix multiplication 
-        implementation with a SYCL blocked matrix multiplication example"
-        LANGUAGES CXX)
+find_package(OpenMP QUIET)
+if(OpenMP_CXX_FOUND)
+    target_link_libraries(matrix_multiply_omp_compare PUBLIC OpenMP::OpenMP_CXX)
+elseif(CMAKE_CXX_COMPILER MATCHES "clang")
```

Comment:
Will this work if the user has set `icpx`? To my knowledge that's what we should be expecting from now on

Reply:
`OpenMP_CXX_FOUND` / `OpenMP_FOUND` are set to true when using `icpx` and the project links the oneAPI OpenMP libs correctly (`/opt/intel/oneapi/compiler/2024.0/lib/libiomp5.so`).

More specifically:
* icpx:
```
-- Found OpenMP_C: -fopenmp (found version "4.5") 
-- Found OpenMP_CXX: -fiopenmp  
-- Found OpenMP: TRUE (found version "4.5")  
```
* clang++ (both from DPC++ or upstream llvm)
```
-- Found OpenMP_C: -fopenmp (found version "4.5") 
-- Could NOT find OpenMP_CXX (missing: OpenMP_CXX_FLAGS OpenMP_CXX_LIB_NAMES) 
-- Could NOT find OpenMP (missing: OpenMP_CXX_FOUND) (found version "4.5")
```

Reply:
Oh, great! :+1: